JamfSync/Model/SynchronizationProgress.swift

```diff
@@ -103,8 +104,15 @@ class SynchronizationProgress: ObservableObject {
         self.currentTotalSizeTransferred = currentTotalSizeTransferred
     }
 
-    private func setFileTransferVars(totalBytesTransferred: Int64, bytesTransferred: Int64) {
-        currentFileSizeTransferred = totalBytesTransferred
+    private func setFileTransferVars(totalBytesTransferred: Int64, bytesTransferred: Int64, progress: SynchronizationProgress) {
+        if progress.operation == "Downloading" {
+            currentFileSizeTransferred = totalBytesTransferred
+            if isAt100Percent() {
+                currentFileSizeTransferred = 0
+            }
+        } else {
+            currentFileSizeTransferred! += bytesTransferred
```

Comment:
You should almost always avoid using an exclamation point. If this theoretically should always be non-nil, then replace ! with ?. If it is nil for some reason, it won't work right since it will ignore the assignment, but at least it won't crash the program.

Reply:
replaced ! with ?